contract/views/contract_portal_templates.xml

```diff
@@ -227,56 +227,87 @@
-                                            </td>
-                                            <td name="td_price_unit" class="text-right">
-                                                <span
-                                                    t-field="line.price_unit"
-                                                    t-options='{"widget": "monetary", "display_currency": contract.currency_id}'
-                                                />
-                                            </td>
-                                            <td
-                                                name="td_recurring_interval"
-                                                class="text-right"
+                                        <tr
+                                            t-att-class="'bg-200 font-weight-bold o_line_section' if line.display_type == 'line_section' else 'font-italic o_line_note' if line.display_type == 'line_note' else ''"
+                                        >
+                                            <t t-if="not line.display_type">
+                                                <td name="td_name">
+                                                    <span t-field="line.name" />
+                                                </td>
+                                                <td
+                                                    name="td_quantity"
+                                                    class="text-right"
+                                                >
+                                                    <span t-field="line.quantity" />
+                                                </td>
+                                                <td
+                                                    name="td_price_unit"
+                                                    class="text-right"
+                                                >
+                                                    <span
+                                                        t-field="line.price_unit"
+                                                        t-options='{"widget": "monetary", "display_currency": contract.currency_id}'
+                                                    />
+                                                </td>
+                                                <td
+                                                    name="td_recurring_interval"
+                                                    class="text-right"
+                                                >
+                                                    <span
+                                                        t-field="line.recurring_interval"
+                                                    />
+                                                    <t
+                                                        t-if="line.recurring_rule_type=='daily'"
+                                                    >Day(s)</t>
```

Comment:
I think this can be obtained programatically instead of having to manually re-map the values to their display name.

Reply:
> I think this can be obtained programatically instead of having to manually re-map the values to their display name.

This lines are not changed (indent due to <t> introduction)

Reply:
This is a copy paste of what is done on sale level

Reply:
> OK, but it can still apply, and simplify a bit the diff of the PR. Something like `line._fields['recurring_rule_type'].selection[line.recurring_rule_type]` or similar if I don't remember bad.

Ok, I got what you mean. This is not the aim of this PR as that code was there before.

The aim here is to avoid products and quantities description when you want to display a note or a section

Reply:
I know, but it was for avoiding huge diff. Do you want me to propose the other imp in a separate PR and then rebase over it?

Reply:
Before :

![image](https://user-images.githubusercontent.com/19529533/116735245-898d9500-a9ee-11eb-83a7-6c605b4fbe3f.png)

After :

![image](https://user-images.githubusercontent.com/19529533/116735433-c8234f80-a9ee-11eb-9372-9d984f909b25.png)